source/Account.cpp

```diff
@@ -115,29 +113,87 @@ void Account::Save(DataWriter &out) const
 
 
 
-// How much the player currently has in the bank.
 int64_t Account::Credits() const
 {
 	return credits;
 }
 
 
 
-// Give the player credits (or pass  negative number to subtract). If subtracting,
-// the calling function needs to check that this will not result in negative credits.
+void Account::SetCredits(int64_t value)
+{
+	credits = value;
+}
+
+
+
 void Account::AddCredits(int64_t value)
 {
 	credits += value;
 }
 
 
 
-// Pay down extra principal on a mortgage.
+int Account::CreditScore() const
+{
+	return creditScore;
+}
+
+
+
+const vector<int64_t> &Account::History() const
+{
+	return history;
+}
+
+
+
+int64_t Account::NetWorth() const
+{
+	return history.empty() ? 0 : history.back();
+}
+
+
+
+void Account::AddHistory(int64_t amount) {
+	history.push_back(amount);
+	if(history.size() > HISTORY)
```

Comment:
should be named MAX_HISTORY_SIZE or somth like that

Reply:
Something like MAX_HISTORY_ENTRIES indeed seems more clear than just HISTORY.

Reply:
also history could be a queue, would fit the nature of adding at the back and removing the front